R/messages.R

```diff
@@ -1,13 +1,19 @@
 messages <- list(
-  errorWrongType = function(objectName, optionalMessage = NULL) {
+  errorWrongType = function(objectName, expectedType, optionalMessage = NULL) {
     # Name of the calling function
     callingFunctions <- sys.calls()
     callingFunction <- sys.call(-length(callingFunctions) + 1)[[1]]
 
+    expectedTypeMsg <- expectedType[1]
+    if (length(expectedType) > 1) {

```

Comment:
I think you can pass a separator to paste0 directly. No need to iterate through each value here.